shell/platform/linux/fl_platform_plugin.cc

```diff
@@ -0,0 +1,172 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/fl_platform_plugin.h"
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_json_method_codec.h"
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_method_channel.h"
+
+#include <gtk/gtk.h>
+
+static constexpr char kChannelName[] = "flutter/platform";
+static constexpr char kBadArgumentsError[] = "Bad Arguments";
+static constexpr char kUnknownClipboardFormatError[] =
+    "Unknown Clipboard Format";
+static constexpr char kClipboardRequestError[] = "Clipboard Request Failed";
+static constexpr char kFailedError[] = "Failed";
+static constexpr char kGetClipboardDataMethod[] = "Clipboard.getData";
+static constexpr char kSetClipboardDataMethod[] = "Clipboard.setData";
+static constexpr char kSystemNavigatorPopMethod[] = "SystemNavigator.pop";
+static constexpr char kTextKey[] = "text";
+
+static constexpr char kTextPlainFormat[] = "text/plain";
+
+struct _FlPlatformPlugin {
+  GObject parent_instance;
+
+  FlMethodChannel* channel;
+};
+
+G_DEFINE_TYPE(FlPlatformPlugin, fl_platform_plugin, G_TYPE_OBJECT)
+
+// Called when clipboard text received.
+static void clipboard_text_cb(GtkClipboard* clipboard,
+                              const gchar* text,
+                              gpointer user_data) {
+  g_autoptr(FlMethodCall) method_call = FL_METHOD_CALL(user_data);
+
+  if (text == nullptr) {
+    g_autoptr(GError) error = nullptr;
+    if (!fl_method_call_respond_error(
+            method_call, kClipboardRequestError,
+            "Failed to retrieve clipboard text from GTK", nullptr, &error))
+      g_warning("Failed to send method call response: %s", error->message);
+    return;
+  }
+
+  g_autoptr(FlValue) result = fl_value_new_map();
+  fl_value_set_string_take(result, kTextKey, fl_value_new_string(text));
+
+  g_autoptr(GError) error = nullptr;
+  if (!fl_method_call_respond_success(method_call, result, &error))
+    g_warning("Failed to send method call response: %s", error->message);
```

Comment:
I'm realizing that this boilerplate shows up in every plugin, and it's always the same. It's hard to imagine a case where a plugin could do anything useful with this, or would log anything different. Given that, we should strongly consider putting the logging inside the method, and having it be void without an `error` out param. Or it could still have an optional error param and return a bool, but log, so that in 99+% of cases people could call it without the error param and not looking at the return value, but it would be an option if they really really wanted it.

Reply:
This PR was particularly bad in that it had multiple cases, which I've refactored to resolve. I was wondering if it made sense to know which plugin was having the error, but the reason the response failed probably has nothing to do with the plugin. I'll do a follow up PR removing these errors.

Reply:
You can always log the channel with the internal logging, which is enough to tell what the plugin is.